api/appbean/AppDetail.go

```diff
@@ -100,24 +101,24 @@ type BuildScript struct {
 }
 
 type CdPipelineDetails struct {
-	Name                          string                            `json:"name"` //pipelineName
-	EnvironmentName               string                            `json:"environmentName" `
-	TriggerType                   pipelineConfig.TriggerType        `json:"triggerType" validate:"required"`
-	DeploymentType                pipelineConfig.DeploymentTemplate `json:"deploymentType,omitempty" validate:"oneof=BLUE-GREEN ROLLING CANARY RECREATE"` //
-	DeploymentStrategies          []*DeploymentStrategy             `json:"deploymentStrategies"`
-	PreStage                      *CdStage                          `json:"preStage"`
-	PostStage                     *CdStage                          `json:"postStage"`
-	PreStageConfigMapSecretNames  *CdStageConfigMapSecretNames      `json:"preStageConfigMapSecretNames"`
-	PostStageConfigMapSecretNames *CdStageConfigMapSecretNames      `json:"postStageConfigMapSecretNames"`
-	RunPreStageInEnv              bool                              `json:"runPreStageInEnv"`
-	RunPostStageInEnv             bool                              `json:"runPostStageInEnv"`
-	IsClusterCdActive             bool                              `json:"isClusterCdActive"`
+	Name                          string                                 `json:"name"` //pipelineName
+	EnvironmentName               string                                 `json:"environmentName" `
+	TriggerType                   pipelineConfig.TriggerType             `json:"triggerType" validate:"required"`
+	DeploymentType                chartRepoRepository.DeploymentStrategy `json:"deploymentType,omitempty" validate:"oneof=BLUE-GREEN ROLLING CANARY RECREATE"` //
```

Comment:
DeploymentType is DeploymentStrategy correct ? can we renamed this

Reply:
there is another object named DeploymentStrategy, changed it to DeploymentStrategyType